sources/packages/backend/apps/queue-consumers/src/processors/schedulers/student-application-notifications/student-application-notifications.scheduler.ts

```diff
@@ -0,0 +1,62 @@
+import { InjectQueue, Process, Processor } from "@nestjs/bull";
+import { QueueService } from "@sims/services/queue";
+import { Job, Queue } from "bull";
+import { BaseScheduler } from "../base-scheduler";
+import {
+  InjectLogger,
+  LoggerService,
+  ProcessSummary,
+} from "@sims/utilities/logger";
+import {
+  getSuccessMessageWithAttentionCheck,
+  logProcessSummaryToJobLogger,
+} from "../../../utilities";
+import { QueueNames } from "@sims/utilities";
+import { ApplicationService } from "../../../services";
+import { NotificationService } from "@sims/services/notifications";
+
+@Processor(QueueNames.StudentApplicationNotifications)
+export class StudentApplicationNotificationsScheduler extends BaseScheduler<void> {
+  constructor(
+    @InjectQueue(QueueNames.StudentApplicationNotifications)
+    schedulerQueue: Queue<void>,
+    queueService: QueueService,
+    private readonly applicationService: ApplicationService,
+    private readonly notificationService: NotificationService,
+  ) {
+    super(schedulerQueue, queueService);
+  }
+
+  @Process()
+  async StudentApplicationNotifications(job: Job<void>): Promise<string[]> {
+    const processSummary = new ProcessSummary();
+
+    try {
+      this.logger.log(
+        `Processing student application notifications job. Job id: ${job.id} and Job name: ${job.name}.`,
+      );
+
+      const eligibleApplications =
```

Comment:
The method `this.applicationService.getEligibleApplicationsForNotification()` will likely be called from another service responsible for processing the scheduler logic.

Reply:
I will wait to see the next PR to have a better context.
Please keep in mind that this PR is to send one notification based on some criteria and that we are expecting other application-based notifications to be created in the future.